src/service/dashboards/reports.rs

```diff
@@ -355,66 +373,70 @@ impl Report {
                 &self.name,
             )
             .await?;
-            self.send_email(&report.0, report.1).await
+            send_email(self, &report.0, report.1).await
         }
     }
+}
 
-    /// Sends emails to the [`Report`] recipients. Currently only one pdf data is supported.
-    async fn send_email(&self, pdf_data: &[u8], dashb_url: String) -> Result<(), anyhow::Error> {
-        let cfg = get_config();
-        if !cfg.smtp.smtp_enabled {
-            return Err(anyhow::anyhow!("SMTP configuration not enabled"));
-        }
-
-        let mut recipients = vec![];
-        for recipient in &self.destinations {
-            match recipient {
-                ReportDestination::Email(email) => recipients.push(email),
-            }
-        }
+/// Sends emails to the [`Report`] recipients. Currently only one pdf data is supported.
+async fn send_email(
```

Comment:
The diff looks complicated but I just moved this function out of the `impl` block and renamed the `self` param to `report`